pmd-core/src/main/java/net/sourceforge/pmd/PropertyDescriptor.java

```diff
@@ -32,12 +61,15 @@
      */
     String description();
 
+
     /**
-     * Denotes the value datatype.
+     * Denotes the value datatype. For multi value properties, this is not the
+     * List class but the list's component class.
      *
-     * @return Class
+     * @return Class literal of the value type
      */
-    Class<T> type();
+    Class<?> type();
```

Comment:
No, for multi valued properties, `T` is the `List<V>`, and the `type()` method returns a `Class<V>`. That's also why I named `V` the type argument in `AbstractMultiValuePropertyDescriptor`, so as not to mistake the two. See [this comment above](https://github.com/pmd/pmd/pull/479#commitcomment-22769778)